tests/acceptance/17_packages/10_new/unsafe/the_great_package_test_generator.py

```diff
@@ -0,0 +1,542 @@
+        ( "32_bit", "1" ): True,
+        ( "32_bit", "2" ): False,
+    },
+    "32_bit_2": {
+        ( "64_bit", "1" ): False,
+        ( "64_bit", "2" ): False,
+        ( "32_bit", "1" ): False,
+        ( "32_bit", "2" ): True,
+    },
+    "64_bit_1_32_bit_1": {
+        ( "64_bit", "1" ): True,
+        ( "64_bit", "2" ): False,
+        ( "32_bit", "1" ): True,
+        ( "32_bit", "2" ): False,
+    },
+    # These combinations are not possible, because different architecture
+    # packages must be the same version.
+    # "64_bit_1_32_bit_2": {
+    #     ( "64_bit", "1" ): True,
+    #     ( "64_bit", "2" ): False,
+    #     ( "32_bit", "1" ): False,
+    #     ( "32_bit", "2" ): True,
+    # },
+    # "64_bit_2_32_bit_1": {
+    #     ( "64_bit", "1" ): False,
+    #     ( "64_bit", "2" ): True,
+    #     ( "32_bit", "1" ): True,
+    #     ( "32_bit", "2" ): False,
+    # },
+    "64_bit_2_32_bit_2": {
+        ( "64_bit", "1" ): False,
+        ( "64_bit", "2" ): True,
+        ( "32_bit", "1" ): False,
+        ( "32_bit", "2" ): True,
+    },
+}
+
+
+def header(test_count):
+    print('''# THIS IS AN AUTOGENERATED TEST!
+# DO NOT EDIT IT DIRECTLY!
+#
+# Instead, edit the_great_package_test_generator.py and use that to regenerate
+# the test.
+#
+# Number of test cases: ''' + str(test_count) + '''
+#
+# If you want to run a specific test case, define a class with the name of that
+# test, for example "from_absent_to_absent___promise_policy_absent_arch_64_bit".
+
+body common control
+{
+    inputs => { "../../../dcs.cf.sub",
+                "../../../../../$(sys.local_libdir)/packages.cf",
+                "../../../../../$(sys.local_libdir)/commands.cf",
+                "../../packages-info.cf.sub",
+              };
+    bundlesequence => { default($(this.promise_filename)) };
+    package_module => "apt_get";
+}
+
+body package_module apt_get
+{
+    query_installed_ifelapsed => "0";
+    query_updates_ifelapsed => "0";
+}
+
+bundle agent init
+{
+  meta:
+      # TODO: Get RedHat support as well.
+      "test_skip_needs_work" string => "!debian";
+}
+
+bundle agent log_test_case(msg)
+{
+  reports:
+      "-------------------------------------"
+        comment => "$(msg)_1";
+      "$(msg)"
+        comment => $(msg);
+      "-------------------------------------"
+        comment => "$(msg)_2";
+}
+''')
+
+def get_possible_promises():
+    # All the possible promises, expressed in terms of policy, architecture,
+    # three versions (1, 2 or latest), and file to install.
+    # P     = policy => "present"
+    # D     = policy => "absent"
+    # F     = File install
+    # R     = Repo install
+    # F64_1 = File to install is 64 bit and version 1
+    # A64   = architecture = "64_bit"
+    # V1    = version => "1"
+    #
+    # This is used to generate the promise structure. It does not include
+    # impossible promises, such as installing a 64-bit file and mentioning
+    # architecture => "32_bit".
```

Comment:
Can this, in its turn, be generated programmatically ?
Doesn't look obviously like it can; but that may just be because I have no idea what it says !